### ISSUE TITLE
Renderers should pass along hook qualifiers

### ISSUE THREAD

**Issue by TheOtherDude** [ciena-frost/ember-frost-bunsen#335]
If I have a list of bunsen details, I would like to be able to use hook qualifiers to specify each individual detail.

**Comment by sandersky** [comment]
Currently bunsen passes the` hook` property down so I'm not sure what the issue is on this addon. I'm guessing what you want is for the `ember-frost-list` to pass down the list items index to each instance as part of the `hook`?

**Comment by TheOtherDude** [comment]
Yes, I want the index on each element, but as far as I'm aware, `hookQualifiers` are passed separately from `hook`.

If I pass `hookQualifiers` like this:
```
{{frost-bunsen-detail
  bunsenModel=bunsenModel
  bunsenView=bunsenView
  hook=(concat hook '-bunsenDetail')
  hookQualifiers=hookQualifiers
  value=record
}}
```

I get the qualifier passed down to the detail, but it doesn't propagate below the first layer: 
<img width="919" alt="screen shot 2017-01-22 at 12 36 51 am" src="https://cloud.githubusercontent.com/assets/8530858/22181175/ef9c0be2-e03a-11e6-9dca-9d3d4d175139.png">

**Comment by job13er** [comment]
The issue is that if the index is not embedded in te hook, and is instead a hook qualifier (as it should be), then the bunsen component doesn't get it because they `hookQualifiers` property isn't passed down into sub-components. I'm actually thinking it might be nice to change the way we include the bunsen ID to use a qualifier called `bunsenId` instead of making it an extension of the hook. Then someone would grab the element by using something like 

```js
$hook('form-input', {bunsenId: 'foo.bar.baz'})
```

And as long as we use the `extend` helper, and `index` that was already in the qualifiers given to the form (or detail) would still be there too:
```js
$hook('form-input', {bunsenId: 'foo.bar.baz', index: 1})
```

**Comment by notmessenger** [comment]
JIRA: https://agile-jira.ciena.com/browse/FROST-771
